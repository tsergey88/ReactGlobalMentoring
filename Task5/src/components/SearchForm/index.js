export SearchForm from "./SearchForm.jsx";
