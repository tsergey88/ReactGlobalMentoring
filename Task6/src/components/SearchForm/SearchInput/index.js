export SearchInput from "./SearchInput.jsx";
