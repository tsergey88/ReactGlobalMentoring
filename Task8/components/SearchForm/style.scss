.search_form_block {
  > div {
    display: flex;
    justify-content: space-between;
    margin-bottom: 1em;
    color: white;
  }
  h3 {
    color: white;
    margin-bottom: 1em;
  }
}
.search_form_tags {
  display: flex;

  div {
    margin-right: 1em;
  }
}
.search_form_btn {
  display: flex;
}