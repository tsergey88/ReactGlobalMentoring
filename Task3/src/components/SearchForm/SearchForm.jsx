import React from "react";
import { SearchRadioButton } from "./SearchRadioButton";
import { Button } from "../Button";
import { SearchInput } from "./SearchInput";

export class SearchForm extends React.Component {
  render() {
    return (
      <div className="search-form-block">
        <h3>FIND YOUR MOVIE</h3>
        <SearchInput />
        <div>
          <div className="search-form-tags">
            <div>
              <span>SEARCH BY</span>
            </div>
            <div>
              <SearchRadioButton
                title="TITLE"
                id="1"
                name="searchBy"
                defaultChecked
              />
              <SearchRadioButton title="GENRE" id="2" name="searchBy" />
            </div>
          </div>
          <div className="search-form-btn">
            <Button title="SEARCH" />
          </div>
        </div>
      </div>
    );
  }
}
