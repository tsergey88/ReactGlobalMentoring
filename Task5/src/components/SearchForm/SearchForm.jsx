// @flow
import React from "react";
import { connect } from "react-redux";

import { fetchFilms } from "../../actions/FilmsAction";
import { SearchRadioButton } from "./SearchRadioButton";
import { Button } from "../Button";
import { SearchInput } from "./SearchInput";

type Props = {
  fetchFilms: () => {},
  limit: number,
  sortBy: string
};

class SearchForm extends React.Component<Props> {
  handleSubmit = e => {
    e.preventDefault();
    const data = {};
    [...e.currentTarget].forEach(field => {
      if (field.type === "text") {
        data[field.name] = field.value;
      }
      if (field.type === "radio" && field.checked) {
        data[field.name] = field.value;
      }
    });
    const { search, searchBy } = data;
    const { fetchFilms, sortBy, limit } = this.props;
    fetchFilms(search, searchBy, sortBy, limit);
  };

  render() {
    return (
      <form
        className="search-form-block"
        action="javascript:void(0)"
        onSubmit={this.handleSubmit}
      >
        <h3>FIND YOUR MOVIE</h3>
        <SearchInput />
        <div>
          <div className="search-form-tags">
            <div>
              <span>SEARCH BY</span>
            </div>
            <div>
              <SearchRadioButton
                title="TITLE"
                id="1"
                name="searchBy"
                defaultChecked
                value="title"
              />
              <SearchRadioButton
                title="GENRE"
                id="2"
                name="searchBy"
                value="genres"
              />
            </div>
          </div>
          <div className="search-form-btn">
            <Button title="SEARCH" type="submit" />
          </div>
        </div>
      </form>
    );
  }
}

const mapStateToProps = state => ({ ...state.FilmsReducer });

const mapDispatchToProps = { fetchFilms };

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SearchForm);
