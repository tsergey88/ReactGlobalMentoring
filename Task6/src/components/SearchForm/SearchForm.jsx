// @flow
import React from "react";
import { connect } from "react-redux";

import { fetchFilms, changeSearchBy } from "../../actions/FilmsAction";
import { SearchRadioButton } from "./SearchRadioButton";
import { Button } from "../Button";
import { SearchInput } from "./SearchInput";

type Props = {
  fetchFilms: () => {},
  changeSearchBy: () => {},
  limit: number,
  sortBy: string,
  searchBy: string,
  searchValue: string,
  history: *
};

class SearchForm extends React.Component<Props> {
  handleSubmit = () => {
    const {
      fetchFilms,
      sortBy,
      limit,
      searchBy,
      searchValue,
      history
    } = this.props;
    history.push(`/search/${searchValue}`)
    fetchFilms(searchValue, searchBy, sortBy, limit);
  };

  handleChange = checkedValue => {
    this.props.changeSearchBy(checkedValue);
  };

  render() {
    const { searchBy } = this.props;
    return (
      <form
        className="search-form-block"
        action="javascript:void(0)"
        onSubmit={this.handleSubmit}
      >
        <h3>FIND YOUR MOVIE</h3>
        <SearchInput />
        <div>
          <div className="search-form-tags">
            <div>
              <span>SEARCH BY</span>
            </div>
            <div>
              <SearchRadioButton
                title="TITLE"
                id="1"
                name="searchBy"
                isChecked={searchBy === "title"}
                value="title"
                onChange={() => this.handleChange("title")}
              />
              <SearchRadioButton
                title="GENRE"
                id="2"
                name="searchBy"
                isChecked={searchBy === "genres"}
                value="genres"
                onChange={() => this.handleChange("genres")}
              />
            </div>
          </div>
          <div className="search-form-btn">
            <Button title="SEARCH" type="submit" />
          </div>
        </div>
      </form>
    );
  }
}

const mapStateToProps = state => ({ ...state.FilmsReducer });

const mapDispatchToProps = { fetchFilms, changeSearchBy };

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SearchForm);
