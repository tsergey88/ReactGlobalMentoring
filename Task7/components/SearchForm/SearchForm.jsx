// @flow
import React from "react";

import { Router } from "../../server/routes.js";
import { SearchRadioButton } from "./SearchRadioButton";
import { Button } from "../Button";
import SearchInput from "./SearchInput";

import styles from "./style.scss";

type Props = {
  fetchFilms: () => {},
  changeSearchBy: () => {},
  limit: number,
  sortBy: string,
  searchBy: string,
  searchValue: string,
  history: *
};

class SearchForm extends React.Component<Props> {
  handleSubmit = () => {
    const { searchValue } = this.props;
    Router.pushRoute(`/search/${searchValue}`);
  };

  handleChange = checkedValue => {
    this.props.changeSearchBy(checkedValue);
  };

  render() {
    const { searchBy } = this.props;
    return (
      <form
        className={styles.search_form_block}
        action="javascript:void(0)"
        onSubmit={this.handleSubmit}
      >
        <h3>FIND YOUR MOVIE</h3>
        <SearchInput />
        <div>
          <div className={styles.search_form_tags}>
            <div>
              <span>SEARCH BY</span>
            </div>
            <div>
              <SearchRadioButton
                title="TITLE"
                id="searchBy_1"
                name="searchBy"
                isChecked={searchBy === "title"}
                value="title"
                onChange={() => this.handleChange("title")}
              />
              <SearchRadioButton
                title="GENRE"
                id="searchBy_2"
                name="searchBy"
                isChecked={searchBy === "genres"}
                value="genres"
                onChange={() => this.handleChange("genres")}
              />
            </div>
          </div>
          <div className={styles.search_form_btn}>
            <Button title="SEARCH" type="submit" />
          </div>
        </div>
      </form>
    );
  }
}

export default SearchForm;
