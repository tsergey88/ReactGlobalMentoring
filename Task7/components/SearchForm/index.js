import SearchForm from "./SearchForm.jsx";

export default SearchForm;
