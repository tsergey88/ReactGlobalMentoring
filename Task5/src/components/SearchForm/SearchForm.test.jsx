import React from "react";
import { SearchForm } from "./SearchForm";
import { shallow } from "enzyme";

describe("SearchForm component", () => {
  const component = shallow(<SearchForm />);
  it("should be render", () => {
    expect(component).toMatchSnapshot();
  });

  it("should be contain SearchInput", () => {
    expect(component.find("SearchInput")).toHaveLength(1);
  });

  it("should be contain SearchRadioButton", () => {
    expect(component.find("SearchRadioButton")).toHaveLength(2);
  });

  it("should be contain Button", () => {
    expect(component.find("Button")).toHaveLength(1);
  });
});
