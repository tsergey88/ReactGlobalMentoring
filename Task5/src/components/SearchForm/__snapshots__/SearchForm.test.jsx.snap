// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component should be contain Button 1`] = `
<div
  className="search-form-block"
>
  <h3>
    FIND YOUR MOVIE
  </h3>
  <SearchInput />
  <div>
    <div
      className="search-form-tags"
    >
      <div>
        <span>
          SEARCH BY
        </span>
      </div>
      <div>
        <SearchRadioButton
          defaultChecked={true}
          id="1"
          name="searchBy"
          title="TITLE"
        />
        <SearchRadioButton
          id="2"
          name="searchBy"
          title="GENRE"
        />
      </div>
    </div>
    <div
      className="search-form-btn"
    >
      <Button
        title="SEARCH"
      />
    </div>
  </div>
</div>
`;

exports[`SearchForm component should be contain SearchInput 1`] = `
<div
  className="search-form-block"
>
  <h3>
    FIND YOUR MOVIE
  </h3>
  <SearchInput />
  <div>
    <div
      className="search-form-tags"
    >
      <div>
        <span>
          SEARCH BY
        </span>
      </div>
      <div>
        <SearchRadioButton
          defaultChecked={true}
          id="1"
          name="searchBy"
          title="TITLE"
        />
        <SearchRadioButton
          id="2"
          name="searchBy"
          title="GENRE"
        />
      </div>
    </div>
    <div
      className="search-form-btn"
    >
      <Button
        title="SEARCH"
      />
    </div>
  </div>
</div>
`;

exports[`SearchForm component should be contain SearchRadioButton 1`] = `
<div
  className="search-form-block"
>
  <h3>
    FIND YOUR MOVIE
  </h3>
  <SearchInput />
  <div>
    <div
      className="search-form-tags"
    >
      <div>
        <span>
          SEARCH BY
        </span>
      </div>
      <div>
        <SearchRadioButton
          defaultChecked={true}
          id="1"
          name="searchBy"
          title="TITLE"
        />
        <SearchRadioButton
          id="2"
          name="searchBy"
          title="GENRE"
        />
      </div>
    </div>
    <div
      className="search-form-btn"
    >
      <Button
        title="SEARCH"
      />
    </div>
  </div>
</div>
`;

exports[`SearchForm component should be render 1`] = `
<div
  className="search-form-block"
>
  <h3>
    FIND YOUR MOVIE
  </h3>
  <SearchInput />
  <div>
    <div
      className="search-form-tags"
    >
      <div>
        <span>
          SEARCH BY
        </span>
      </div>
      <div>
        <SearchRadioButton
          defaultChecked={true}
          id="1"
          name="searchBy"
          title="TITLE"
        />
        <SearchRadioButton
          id="2"
          name="searchBy"
          title="GENRE"
        />
      </div>
    </div>
    <div
      className="search-form-btn"
    >
      <Button
        title="SEARCH"
      />
    </div>
  </div>
</div>
`;
